_posts/2020-02-07-metal3-dev-env-install-deep-dive.md

```diff
@@ -0,0 +1,921 @@
+total 0
+drwxr-x---. 2 root root 21 Feb  5 11:07 sushy-tools
+drwxr-x---. 4 root root 70 Feb  5 11:08 vbmc
+```
+
+Next, both host provisioning and baremetal interfaces are configured. Provisioning interface, as the name suggests, it will be used to provision the virtual bare metal hosts by means of the `BareMetal operator`. This interface is configured with an static IP (172.22.0.1):
+
+```sh
+[alosadag@smc-master metal3-dev-env]$ ifconfig provisioning
+provisioning: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
+        inet 172.22.0.1  netmask 255.255.255.0  broadcast 172.22.0.255
+        inet6 fe80::1091:c1ff:fea1:6a0f  prefixlen 64  scopeid 0x20<link>
+        ether 12:91:c1:a1:6a:0f  txqueuelen 1000  (Ethernet)
+```
+
+On the other hand, the baremetal virtual interface behaves as an external network. This interface is able to reach the internet and it is the network where the different Kubernetes nodes will exchange information. This interface is configured as auto, so the IP is retrieved by DHCP.
+
+```sh
+[alosadag@smc-master metal3-dev-env]$ ifconfig baremetal
+baremetal: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
+        inet 192.168.111.1  netmask 255.255.255.0  broadcast 192.168.111.255
+        ether 52:54:00:db:85:29  txqueuelen 1000  (Ethernet)
+```
+
+Next, an Ansible role called [firewall](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/firewall.yml) will be executed targetting the host to be sure that the proper ports are opened. In case your host is running `Red Hat Enterprise Linux` or `CentOS 8`, firewalld module will be used. All other cases, iptables module is the choice. 
+
+Below, the code snippet where `firewalld` or `iptables` is assigned:
+
+```sh
+# Use firewalld on CentOS/RHEL, iptables everywhere else
+
+export USE_FIREWALLD=False
+if [[ ($OS == "rhel" || $OS = "centos") && ${OS_VERSION} == 8 ]]
+then
+  export USE_FIREWALLD=True
+fi
+```
+
+> **Note:** This behaviour can be changed by replacing the value of the `USE_FIREWALLD` variable
+<br>
+<br>
+
+The ports managed by this role are all associated to the services that take part of the provisioning process: `ironic`, `vbmc`, `httpd`, `pxe`, `container registry`..
+<br>
+<br>
+> **Note:**  Services like ironic, pxe, keepalived, httpd and the container registry are running in the host as containers attached to the host network on the host's provisioning interface. On the other hand, vbmc service, which is also running as a privileged container, will be listening in the host's baremetal interface.
+<br>
+<br>
+
+Once the network is configured, a local `container registry` is started. It will be needed if we're going to use any locally built images. In that case, container images can be modified locally and pushed to the local registry. At that point, the specific image location variable must be changed so it must point out the local registry. This process makes easy to verify and test changes to the code locally.
```

Comment:
It will be needed in the case of any locally built images usage. Then, the container images can be modified locally and pushed to the local registry.